### ISSUE TITLE
🐛 `is_ground_state` does not use `QuickSim`'s results

### ISSUE THREAD

**Issue by wlambooy** [cda-tum/fiction#392]
### fiction version

latest

### OS

doesn't matter

### Description

It's clear to see what the problem is here:

```c++
template <typename Lyt>
[[nodiscard]] bool is_ground_state(const sidb_simulation_result<Lyt>& quicksim_results,
                                   const sidb_simulation_result<Lyt>& exhaustive_results) noexcept
{
    static_assert(is_cell_level_layout_v<Lyt>, "Lyt is not a cell-level layout");
    static_assert(has_sidb_technology_v<Lyt>, "Lyt is not an SiDB layout");

    if (exhaustive_results.charge_distributions.empty())
    {
        return false;
    }

    const auto min_energy_exact  = minimum_energy(exhaustive_results.charge_distributions.cbegin(),
                                                  exhaustive_results.charge_distributions.cend());
    const auto min_energy_new_ap = minimum_energy(exhaustive_results.charge_distributions.cbegin(),
                                                  exhaustive_results.charge_distributions.cend());

    return round_to_n_decimal_places(std::abs(min_energy_exact - min_energy_new_ap), 6) == 0;
}
```

I got a warning for an unused parameter (`quicksim_results`), so I thought I would put a small fix in the `ClusterComplete` PR, but apparently TTS fails now, which is unrelated. So I took out the change there and made this issue instead, highlighting the problem.

### How to Reproduce

-

**Comment by Drewniok** [comment]
@wlambooy You are absolutely right. This is the bug I mentioned. It is fixed here https://github.com/cda-tum/fiction/pull/380.

**Comment by wlambooy** [comment]
> @wlambooy You are absolutely right. This is the bug I mentioned. It is fixed here #380.

Ah, I recall that you said so! Everything passes now, great!
